src/main/runner/shims/http/index.ts

```diff
@@ -0,0 +1,84 @@
+import http from 'k6/http'
+
+import { instrumentParams } from './utils'
+
+// Store original methods before mutation
+const originalRequest = http.request
+const originalAsyncRequest = http.asyncRequest
+
+interface HttpURL {
+  __brand: 'http-url'
+}
+
+// Mutate the native http module to add instrumentation
+http.request = function <RT extends http.ResponseType | undefined>(
+  method: string,
+  url: string | HttpURL,
+  body?: http.RequestBody | null,
+  params?: http.RefinedParams<RT> | null
+) {
+  return originalRequest<RT>(method, url, body, instrumentParams(params))
+}
```

Comment:
Typescript is able to infer these types correctly for me.

```suggestion
http.request = function (
  method
  url,
  body?,
  params?
) {
  return originalRequest(method, url, body, instrumentParams(params))
}
```